### DISCUSSION TITLE
xstate sharing the model with other machines

### DISCUSSION THREAD

**Discussion by MonobikashDas** [statelyai/xstate#3746] [Q&A]
Can anyone help me to understand how can I share model from one state machine to another?

**Comment by davidkpiano** [comment]
Are you talking about the model from `createModel`?

**Reply by MonobikashDas** [reply]
Yes @davidkpiano . I have created a model using createModel. There is a field which  need to be access in another state machine.

**Reply by davidkpiano** [reply]
The `createModel()` function is being deprecated, and will be removed in XState v5.

But if you are using it, you can create different machines via:

```js
const model = createModel(...);

const machine1 = model.createMachine({ ... }); // some machine
 
const machine2 = model.createMachine({ ... }); // different machine, maybe different states
```

**Reply by MonobikashDas** [reply]
I have 2 different models(model1, model2) created in 2 machines(machine1, machine2). My problem is - i want to access model1 in machine 2.

**Reply by davidkpiano** [reply]
This is not currently possible, or at least pretty tricky to do. You would need to combine the models somehow.
